File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.js
const fs = require('fs')
const filepath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const input = fs.readFileSync(filepath).toString().split('\n');
const n = +input[0];
let arr = input[1].split(' ').map(Number);
const m = +input[2];
let operations = [];
for (let i = 0; i < m; i++) {
    let operation = input[3 + i].split(' ').map(Number);
    operations.push(operation);
}

function changeArr(){
    for(let i = 0; i< m; i++){
        const gender = operations[i][0];
        const num = operations[i][1];
        if(gender == 1){
            for (let j = num; j <= n; j += num)
                {
                    arr[j - 1] = arr[j - 1] === 1 ? 0 : 1;
                }
            }
        else if(gender == 2){
            let index = num - 1;
            arr[index] = arr[index] === 1 ? 0 : 1;
            for (let j = 1; index - j >= 0 && index + j < n; ++j)
            {
                if (arr[index - j] == arr[index + j])
                {
                    arr[index - j] = arr[index - j] === 1 ? 0 : 1;
                    arr[index + j] = arr[index + j] === 1 ? 0 : 1;
                }
                else
                {
                    break;
                }
            }
        }
    }

}

function PrintResult()
{
    let result = '';
    for (let i = 0; i < n; i++) {
        result += arr[i] + ' ';
        if ((i + 1) % 20 === 0) {
            result += '\n';
        }
    }
    console.log(result.trim());
}

changeArr();
PrintResult();

File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.cc
#include <iostream>
#include <vector>
using namespace std;

int main()
{
    int n;
    cin >> n;
    vector <int> v(n);
    for (auto& i : v)
    {
        cin >> i;
    }
    int m;
    cin >> m;
    for (int i = 0; i < m; i++)
    {
        int gender, num;
        cin >> gender >> num;

        if (gender == 1) //남학생의 경우
        {
            int index = num;
            while (index <= n)
            {
                v[index-1] = !v[index-1];
                index += num;
            }
        }
        else //여학생의 경우
        {
            int index = num - 1;
            v[index] = !v[index];
            int i = 1;
            while (index - i >= 0 && index + i < n)
            {
                if (v[index - i] == v[index + i])
                {
                    v[index - i] = !v[index - i];
                    v[index + i] = !v[index + i];
                }
                else
                    break;
                i++;
            }
        }
    }
    for (int i = 0; i < v.size(); i++)
    {
        if (i != 0 && i % 20 == 0)
            cout << '\n';
        cout << v[i] << " ";
    }
    return 0;
}
 